be/src/exec/vectorized/hash_join_node.cpp

```diff
@@ -71,14 +71,26 @@ Status HashJoinNode::init(const TPlanNode& tnode, RuntimeState* state) {
     }
 
     if (tnode.hash_join_node.__isset.partition_exprs) {
+        // the same column can appear more than once in either lateral side of eq_join_conjuncts, but multiple
+        // occurrences are accounted for once when determining local shuffle partition_exprs for bucket shuffle join.
+        // for an example:
+        // table t1 is bucketed by c1, the query 'select * from t1,t2 on t1.c1 = t2.c1 and t1.c1 in (TRUE, NULL) =
+        // t2.c1', SlotRef(t2.c1) appears twice, but the local shuffle partition_exprs should have the same number of
+        // exprs as partition_exprs used by ExchangeNode that sends data to right child of the HashJoin.
         for (const auto& partition_expr : tnode.hash_join_node.partition_exprs) {
+            bool match_exactly_once = false;
             for (auto i = 0; i < eq_join_conjuncts.size(); ++i) {
                 const auto& eq_join_conjunct = eq_join_conjuncts[i];
                 if (eq_join_conjunct.left == partition_expr || eq_join_conjunct.right == partition_expr) {
+                    match_exactly_once = true;
                     _probe_equivalence_partition_expr_ctxs.push_back(_probe_expr_ctxs[i]);
                     _build_equivalence_partition_expr_ctxs.push_back(_build_expr_ctxs[i]);
+                    break;
                 }
             }
+            if (!match_exactly_once) {
+                return Status::InternalError("Invalid partition expr for bucket shuffle join");
```

Comment:
Why you return the error for the query？if the query plan has sent to the BE, BE must execute it right.